packages/lisk-p2p/src/errors/p2p_error.ts

```diff
@@ -0,0 +1,12 @@
+export class P2PError extends Error {
+	private readonly _name: string;
```

Comment:
`name` can be public and exactly same as the class name?

Reply:
Yes, it works and the getter overrides the default name property. It's a tricky case because the `name` property is public on the `Error` class but at the same time we want to discourage public properties.

That approach solves that problem and prevents users from overwriting the name property but I guess we can decide to allow public `name` property in our case for the sake of consistency with JavaScript.